src/core/pagerankGraph.test.js

```diff
@@ -63,6 +69,50 @@ describe("core/pagerankGraph", () => {
     });
   });
 
+  describe("node prefix filter matches graph filter", () => {
+    const n1 = NodeAddress.empty;
+    const n2 = NodeAddress.fromParts(["foo"]);
+    const n3 = NodeAddress.fromParts(["foo", "bar"]);
+    const n4 = NodeAddress.fromParts(["zod", "bar"]);
+    const g = () =>
+      new Graph()
+        .addNode(n1)
+        .addNode(n2)
+        .addNode(n3)
+        .addNode(n4);
+    const pg = () => new PagerankGraph(g(), defaultEvaluator);
+
+    function expectPagerankGraphToEqualGraph(
+      options: {|+prefix: NodeAddressT|} | void
+    ) {
+      const pagerankGraphNodes = Array.from(pg().nodes(options)).sort();
+      const graphNodes = Array.from(g().nodes(options)).sort();
+
+      pagerankGraphNodes.forEach(
+        (pgNode, i) =>
+          expect(pgNode.node).toEqual(graphNodes[i]) &&
+          expect(pgNode.score).toBe(0.25)
+      );
+    }
+
+    it("with no options object", () => {
+      expectPagerankGraphToEqualGraph(undefined)
+    });
+
+    it("requires a prefix when options are specified", () => {
+      // $ExpectFlowError
+      expect(() => Array.from(pg().nodes({}))).toThrow("prefix");
+    });
+
+    it("with prefix filter", () => {
+      expectPagerankGraphToEqualGraph({prefix: n2});
+    });
+
+    it("with prefix that matches nothing", () => {
```

Comment:
For completeness, let's also check with the prefix `NodeAddress.empty` (which matches everything).

Reply:
In a separate PR, ought I add that test to `Graph.test` too?

Reply:
Sure. I would be fine with doing it in the same PR too, tbh.